### ISSUE TITLE
RFC: DROPBEAR_ALWAYS_DELAY_HOSTKEY build option

### ISSUE THREAD

**Issue by ryancdotorg** [mkj/dropbear#221]
I am maintaining a patch that I'd like to cleanup and have included.

The cleaned up version would add a build option, `DROPBEAR_ALWAYS_DELAY_HOSTKEY`, which does the following:

* Removes `svr_opts.delay_hostkey` and acts as if it's always set to 1
* Ignores the `-R` flag to dropbear server (for compatibility)
* When dropbear is being built as a multicall binary without client support, changes dropbearkey to do nothing more than print 'does nothing, successfully' to stderr and exit

Would this be something that could be merged if I submit a PR?

**Comment by mkj** [comment]
What's the reason for having it as a build option rather than modifying the startup scripts? I'm not totally opposed, but it seems like a bit of a mess if the same thing got added for every commandline option.

**Comment by ryancdotorg** [comment]
For a multi-call binary, omitting `dropbearkey` saves up to 24Kb depending on architecture and which programs are included.

The savings is less if `dropbearconvert` and/or `scp` is being included.

I have some hardware that runs a vendor-modified version of OpenWRT which includes a version of dropbear patched to only run the vendor's configuration CLI. My patch allows drop-in replacement of their binary without having to modify the init scripts or any configuration.

Basically, it's a convenience.

**Comment by mkj** [comment]
I'm not sure it's worth adding just to avoid modifying init scripts.
If you really want to avoid touching /etc, make dropbear a script

```sh
#!/bin/sh
exec dropbear.real -R "$@"
```

You can save the compile size by omitting dropbearconvert and dropbearkey entirely from the multi binary:

```sh
make PROGRAMS="dropbear dbclient" MULTI=1
```

**Comment by ryancdotorg** [comment]
The script is a good idea. I will probably compile without `dropbearkey`, drop a dispatch script as `/usr/bin/dropbear`,  then drop the binary at `/usr/bin/dropbearmulti`. Feel free to close this.

```sh
#!/bin/sh
if [ "$0" = "dropbear" ]; then
    exec dropbearmulti dropbear -R "$@"
elif [ "$0" = "dropbearkey" ]; then
    exit 0
else
    exec dropbearmulti "$0" "$@"
fi
```
